src/lib/resolvers/partialDMChannel.ts

```diff
@@ -9,7 +9,8 @@ export function resolvePartialDMChannel(
 	message: Message
 ): Result<DMChannel | PartialDMChannel, Identifiers.ArgumentChannelError | Identifiers.ArgumentDMChannelError> {
 	const result = resolveChannel(parameter, message);
-	if (!result.success) return result;
-	if (isDMChannel(result.value)) return ok(result.value);
-	return err(Identifiers.ArgumentDMChannelError);
+	if (!result.isOk()) return result;
+	const channel = result.unwrap();
+	if (isDMChannel(channel)) return Result.ok(channel);
+	return Result.err(Identifiers.ArgumentDMChannelError);
```

Comment:
I'm highly considering `mapOkFlat` and `mapErrFlat` for this, since this pattern is repeated quite a lot.